extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -141,15 +141,23 @@ public class VertxHttpRecorder {
 
     public static final String GET = "GET";
     private static final Handler<HttpServerRequest> ACTUAL_ROOT = new Handler<HttpServerRequest>() {
+
+        /** JVM system property that disables URI validation, don't use this in production. */
+        private static final String DISABLE_URI_VALIDATION_PROP_NAME = "vertx.disableURIValidation";
+
+        /**
+         * Constant that disables HTTP headers validation, this is a constant so the JIT can eliminate validation code
+         * and save some allocations.
+         */
+        private final boolean DISABLE_URI_VALIDATION = Boolean.getBoolean(DISABLE_URI_VALIDATION_PROP_NAME);
```

Comment:
must be static too in order to be treated as constant

Reply:
👍🏻 Good catch!

Reply:
I thought of that, but decided against it so as to not have it captured during native image build

Reply:
Makes sense, but still the comment
```
this is a constant so the JIT can eliminate validation code and save some allocations.
```
is not valid anymore
Netty is using many system properties like these and I expect them to be correctly handled on native image (which knowledge I'm still lacking) - is it not the same?

Reply:
The constants we care about to not be captured in native image, we explicitly "reset" using a substitution.

We could of course do that here too, but it feels like too much work for such a niche feature

Reply:
At this point let's treat this as a (non-JIT) final field that save some useless work (that's fine)

Reply:
Good point, however the handler is only created once :).

I did push an update to the comment